bftengine/tests/simpleTest/config_file_parser_test.cpp

```diff
@@ -0,0 +1,66 @@
+#include "config_file_parser.hpp"
+#include "../../src/bftengine/assertUtils.hpp"
+
+#include <iostream>
+#include <string>
+
+using std::cout;
+using std::cin;
+using std::string;
+using std::vector;
+
+int main(int argc, char **argv) {
+
+  cout << "Enter configuration file name with a full path:\n";
+
+  const string default_config_file = "sample_config.txt";
+  string config_file = default_config_file;
+  string given_config_file;
+  const string use_default_config_file = ".\n";
+  const uint expected_replicas_num = 4;
+  const uint expected_clients_num = 1;
+  const string expected_replica1 = "127.0.0.1:3410";
+  const string expected_replica2 = "127.0.0.1:3420";
+  const string expected_replica3 = "127.0.0.1:3430";
+  const string expected_replica4 = "127.0.0.1:3440";
+  const string expected_client   = "127.0.0.1:4444";
+  const string values_to_split = "10.23.43.1:1234:1238";
+  const string expected_split_values[] = { "10.23.43.1", "1234", "1238"};
+  const string values_to_split_delimiter = ":";
+
+  cin >> given_config_file;
```

Comment:
I don't think this is actually working. It seems to be trying to open a local sample_config.txt no matter what.

```
~/builds/concord-bft/debug/bftengine/tests/simpleTest$ ./config
Enter configuration file name with a full path:
/home/bfink/vmwathena/concord-bft/bftengine/tests/simpleTest/sample_config.txt
FATAL 2019-02-26 10:56:22.520 (simpletest.test) Failed to open file: sample_config.txt
```

If I copy `sample_config.txt` into the local directory, it successfully opens that file no matter what path I type in.

Ultimately, my question is, should this be a parameter that is typed in? Given the new validation logic, the test is tightly coupled to the example file. Maybe that path should just be static?

Reply:
> @yuliasherman, VMware has approved your signed contributor license agreement.

As agreed with Guy, we leave our *.sh scripts without changes because the intention is to move to python scripts. testReplicasAndClients.sh does not support configuration on remote computers anyway. Our python scripts should be also modified to start supporting launching replicas on remote nodes. This should be done as a different task.

Reply:
You are right, I forgot to test this case - when a given file is different from the default. That's exactly a reason why I want to leave this option supported - it verifies that a given file is used and not a default one.